app/lib/actions/katello/capsule_content/create_or_update.rb

```diff
@@ -0,0 +1,45 @@
+module Actions
+  module Katello
+    module CapsuleContent
+      class CreateOrUpdate < ::Actions::EntryAction
+        # @param capsule_content [::Katello::CapsuleContent]
+        # @param pulp_repo [::Katello::Glue::Pulp::Repo]
+        def plan(capsule_content, environment = nil, content_view = nil)
+          fail _("Action not allowed for the default capsule.") if capsule_content.default_capsule?
+
+          current_repos_on_capsule = capsule_content.current_repositories(environment, content_view)
+          list_of_repos_to_sync = capsule_content.pulp_repos(environment, content_view)
+          need_creation =  list_of_repos_to_sync - current_repos_on_capsule
+          need_update = current_repos_on_capsule & list_of_repos_to_sync
+
+          need_creation.each do |repo|
+            create_repo_in_pulp(capsule_content, repo)
+          end
+
+          need_update.each do |repo|
+            plan_action(Pulp::Repository::Refresh, repo, capsule_id: capsule_content.capsule.id)
+          end
+        end
+
+        def create_repo_in_pulp(capsule_content, repository)
+          ueber_cert = ::Katello::Cert::Ca.ueber_cert(repository)
+          relative_path = repository.puppet? ? nil : repository.relative_path
```

Comment:
actually relative_path is not nil if its a content_view_environment (in which case its the install_path) if i remember correctly:

https://github.com/Katello/katello/blob/master/app/lib/actions/katello/content_view_puppet_environment/create.rb#L18